libgalois/src/analytics/k_truss/k_truss.cpp

```diff
@@ -0,0 +1,459 @@
+            katana::FindEdgeSortedByDest(*g, e.second, e.first)) = removed;
+      }
+    }
+  };
+
+  void operator()(Graph* g, unsigned int k) {
+    if (k - 2 == 0) {
+      return;
+    }
+
+    EdgeVec work[2];
+    EdgeVec *cur = &work[0], *next = &work[1];
+    size_t curSize, nextSize;
+
+    //! Symmetry breaking:
+    //! Consider only edges (i, j) where i < j.
+    katana::do_all(
+        katana::iterate(*g),
+        [&g, cur](GNode n) {
+          for (auto e : g->edges(n)) {
+            auto dest = g->GetEdgeDest(e);
+            if (*dest > n) {
+              cur->push_back(std::make_pair(n, *dest));
+            }
+          }
+        },
+        katana::steal());
+    curSize = std::distance(cur->begin(), cur->end());
+
+    //! Remove unsupported edges until no more edges can be removed.
+    while (true) {
+      katana::do_all(
+          katana::iterate(*cur), KeepSupportedEdges{g, k - 2, *next},
+          katana::steal());
+      nextSize = std::distance(next->begin(), next->end());
+
+      if (curSize == nextSize) {
+        //! Every edge in *cur is kept, done
+        break;
+      }
+
+      cur->clear();
+      curSize = nextSize;
+      std::swap(cur, next);
+    }
+  }  ///< End operator()
+};   ///< End struct BSPTrussAlgo
+
+/**
+ * BSPCoreAlgo:
+ * 1. Keep nodes w/ degree >= k and remove all edges for nodes whose degree < k.
+ * 2. If all nodes are kept, done.
+ * 3. Go back to 1.
+ */
+struct BSPCoreAlgo {
+  std::string name() { return "bspCore"; }
+
+  struct KeepValidNodes {
+    Graph* g;
+    unsigned int j;
+    NodeVec& s;
+
+    KeepValidNodes(Graph* g, unsigned int j, NodeVec& s) : g(g), j(j), s(s) {}
+
+    void operator()(GNode n) {
+      if (IsValidDegreeNoLessThanJ(*g, n, j)) {
+        s.push_back(n);
+      } else {
+        for (auto e : g->edges(n)) {
+          auto dest = g->GetEdgeDest(e);
+          g->template GetEdgeData<EdgeFlag>(
+              katana::FindEdgeSortedByDest(*g, n, *dest)) = removed;
+          g->template GetEdgeData<EdgeFlag>(
+              katana::FindEdgeSortedByDest(*g, *dest, n)) = removed;
+        }
+      }
+    }
+  };
+
+  void operator()(Graph* g, unsigned int k) {
+    NodeVec work[2];
+    NodeVec *cur = &work[0], *next = &work[1];
+    size_t curSize = g->num_nodes(), nextSize;
+
+    katana::do_all(
+        katana::iterate(*g), KeepValidNodes{g, k, *next}, katana::steal());
+    nextSize = std::distance(next->begin(), next->end());
+
+    while (curSize != nextSize) {
+      cur->clear();
+      curSize = nextSize;
+      std::swap(cur, next);
+
+      katana::do_all(
+          katana::iterate(*cur), KeepValidNodes{g, k, *next}, katana::steal());
+      nextSize = std::distance(next->begin(), next->end());
+    }
+  }
+};  ///< End BSPCoreAlgo.
+
+/**
+ * BSPCoreThenTrussAlgo:
+ * 1. Reduce the graph to k-1 core
+ * 2. Compute k-truss from k-1 core
+ */
+struct BSPCoreThenTrussAlgo {
+  std::string name() { return "bspCoreThenTruss"; }
```

Comment:
Nit: `name()` is never used. And if that's removed then we don't need the struct and can use a real function instead.